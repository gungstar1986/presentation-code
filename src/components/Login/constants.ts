import * as Yup from 'yup';

import { LoginData } from 'components/Login/interfaces';

export const loginFormikData: LoginData = {
  email: '',
  password: '',
  rememberMe: false,
};

const passValidate = {
  numbers: /(?=.*[0-9])/,
  lowercase: /(?=.*[a-z])/,
  uppercase: /(?=.*[A-Z])/,
  symbols: /(?=.*[*@#$%!&])/,
  length: /[0-9a-zA-Z*@#$%!&]{8,}/,
};

export const loginFormicSchema = Yup.object({
  email: Yup.string().email('Incorrect field').required('Required field'),
  password: Yup.string()
    .matches(passValidate.lowercase, 'At least 1 symbol in lower case')
    .matches(passValidate.uppercase, 'At least 1 symbol in upper case')
    .matches(passValidate.numbers, 'At least 1 number')
    .matches(passValidate.symbols, 'At least 1 service symbol')
    .matches(passValidate.length, 'Length must be more then 8 symbols')
    .required('Required field'),
});
