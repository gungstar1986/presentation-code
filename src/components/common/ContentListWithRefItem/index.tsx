import React, { useRef, useState } from 'react';

import cn from 'classnames';

import { scrollToTop } from 'helpers/scrollTo';
import { createRefElem } from 'helpers/createRefElem';
import { ContentListProps } from 'components/common/ContentListWithRefItem/interfaces';

export const ContentListWithRefItem: React.FC<ContentListProps> = ({ content, loadFn, contentItem }) => {
  const [scroll, setScroll] = useState<boolean>(true);

  const activeEl = useRef(null);

  const handleShowContentItem = (id: number) => {
    if (contentItem.id === id) return;
    loadFn(id);
  };

  const list = content.map((beerItem, index) => {
    const { name, id, image_url } = beerItem;

    const active = id === contentItem?.id;
    const lastElem = index === content.length - 1;
    const className = cn('content-link-container', { active });

    if (lastElem && scroll) scrollToTop({ index, content, ref: activeEl, setScroll });

    return createRefElem(active, { id, className, handleShowContentItem, image: image_url, name, ref: activeEl });
  });

  return <>{list}</>;
};
