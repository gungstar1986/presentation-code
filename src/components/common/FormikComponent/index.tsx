import React, { createContext } from 'react';

import { useCustomFormik } from 'components/common/FormikComponent/hooks/useCustomFormik';
import { FormikContextInterface, FormikContextProps } from './interfaces';

export const FormikContext = createContext<FormikContextInterface>(null);

export const FormikContextProvider: React.FC<FormikContextProps> = ({
  accept,
  reject,
  formikData,
  formikSchema,
  children,
}) => {
  const formik = useCustomFormik({ values: formikData, schema: formikSchema, onSubmit: accept });
  const { values, errors, touched, handleChange, handleSubmit } = formik;

  const store: FormikContextInterface = { values, errors, touched, handleChange, reject };

  return (
    <FormikContext.Provider value={store}>
      <form onSubmit={handleSubmit}>{children}</form>
    </FormikContext.Provider>
  );
};
