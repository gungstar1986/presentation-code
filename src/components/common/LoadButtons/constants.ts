export const PREV_BUTTON = 'Load Prev';
export const NEXT_BUTTON = 'Load Next';
