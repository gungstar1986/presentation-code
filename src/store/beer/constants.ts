export const START_PAGE = 0;
export const PER_PAGE = 10;
export const MIN_CONTENT_ITEMS = 1;
export const MAX_CONTENT_ITEMS = 325;

export const paginate = {
  prev: START_PAGE,
  curr: START_PAGE,
  next: START_PAGE,
  per_page: PER_PAGE,
};
