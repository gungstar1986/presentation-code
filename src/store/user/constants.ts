export const EMAIL = 'example@example.com';
export const PASSWORD = 'Test123456*';

export const USER = {
  name: 'Example',
  role: 'User',
  email: EMAIL,
  password: PASSWORD,
};
