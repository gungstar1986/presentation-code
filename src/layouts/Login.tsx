import React from 'react';
import { observer } from 'mobx-react-lite';

import { LoginData } from 'components/Login/interfaces';
import { SimpleInput } from 'components/common/FormikComponent/Inputs/SimpleInput';
import { CheckboxInput } from 'components/common/FormikComponent/Inputs/CheckboxInput';
import { useFormikContext } from 'components/common/FormikComponent/hooks/useFormikContext';

// to using standard styles for input components uncomment the line below
import 'components/common/FormikComponent/styles/styles.scss';

export const Login: React.FC = observer(() => {
  const { handleChange, errors, touched, values } = useFormikContext();

  const { email, password, rememberMe } = values as LoginData;

  return (
    <div className="login">
      <div className="login-body-item">
        <div className="form-control">
          <SimpleInput
            name="email"
            type="text"
            title="Email"
            value={email}
            errors={errors}
            touched={touched}
            className="input"
            placeholder="Email"
            inputLabel="input-label"
            classNameError="input-label error"
            handleChange={handleChange}
          />
        </div>

        <div className="form-control">
          <SimpleInput
            name="password"
            type="password"
            title="password"
            value={password}
            errors={errors}
            touched={touched}
            className="input"
            placeholder="Password"
            inputLabel="input-label"
            classNameError="input-label error"
            handleChange={handleChange}
          />
        </div>

        <CheckboxInput
          name="rememberMe"
          title="Remember me?"
          className="checkbox-input"
          errors={errors}
          touched={touched}
          handleChange={handleChange}
          defaultChecked={rememberMe}
        />

        <button className="btn-primary mod" type="submit">
          Вход
        </button>
      </div>
    </div>
  );
});
