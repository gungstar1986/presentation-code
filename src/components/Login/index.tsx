import React from 'react';
import { observer } from 'mobx-react-lite';

import { useUserStore } from 'hooks/storesHooks/useUserStore';
import { useRedirectToCabinet } from 'hooks/customHooks/useRedirectToCabinet';

import { Login } from 'layouts/Login';
import { User } from 'store/user/interfaces';
import { FormikContextProvider } from 'components/common/FormikComponent';
import { loginFormicSchema, loginFormikData } from 'components/Login/constants';

export const LoginMainComponent: React.FC = observer(() => {
  const { checkUserIsAuth } = useUserStore();

  useRedirectToCabinet();

  const handleSubmit = (data: User) => {
    checkUserIsAuth(data);
  };

  return (
    <div className="main-content-container">
      <div className="form-container">
        <FormikContextProvider accept={handleSubmit} formikData={loginFormikData} formikSchema={loginFormicSchema}>
          <Login />
        </FormikContextProvider>
      </div>
    </div>
  );
});
