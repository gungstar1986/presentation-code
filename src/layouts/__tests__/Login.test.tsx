import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { Login } from 'layouts/Login';
import { FormikContextProvider } from 'components/common/FormikComponent';
import { loginFormicSchema, loginFormikData } from 'components/Login/constants';

describe('Login component', function () {
  it('should return Login component', function () {
    const handleSubmit = jest.fn();

    const wrapper = render(
      <BrowserRouter>
        <FormikContextProvider accept={handleSubmit} formikData={loginFormikData} formikSchema={loginFormicSchema}>
          <Login />
        </FormikContextProvider>
      </BrowserRouter>
    );

    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
