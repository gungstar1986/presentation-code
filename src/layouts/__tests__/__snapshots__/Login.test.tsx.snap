// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should return Login component 1`] = `
<body>
  <div>
    <form>
      <div
        class="login"
      >
        <div
          class="login-body-item"
        >
          <div
            class="form-control"
          >
            <label
              class="input-label"
              for="email"
            >
              Email
            </label>
            <input
              class="input"
              id="email"
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-control"
          >
            <label
              class="input-label"
              for="password"
            >
              password
            </label>
            <input
              class="input"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="checkbox"
          >
            <input
              class="checkbox-input"
              id="rememberMe"
              name="rememberMe"
              type="checkbox"
            />
            <label
              class="checkbox-label"
              for="rememberMe"
            >
              <span
                class="checkbox-text"
              >
                Remember me?
              </span>
            </label>
          </div>
          <button
            class="btn-primary mod"
            type="submit"
          >
            Вход
          </button>
        </div>
      </div>
    </form>
  </div>
</body>
`;
